### ISSUE TITLE
Subdivision depth

### ISSUE THREAD

**Issue by vinceh121** [stefangabos/world_countries#85]
It seems that some states only have their top-most subdivisions indicated in the `data/subdivisions/*` files. E.g. France only has régions (FR-OCC...) but lacks départements (FR-31...)[^1], Great Britain has countries (GB-ENG...) but lacks counties (GB-ESS)[^2]

Would there be a problem in adding those "deeper" subdivisions?

[^1]: https://en.wikipedia.org/wiki/ISO_3166-2:FR
[^2]: https://en.wikipedia.org/wiki/ISO_3166-2:GB

**Comment by stefangabos** [comment]
yes, i have never paid too much attention to what is in the subdivisions nor am i familiar with other countries' subdivisions. i was hoping for peoples' contributions for the management of subdivisions.
but now that i see that wikipedia has most of the needed info i will write a script to fetch everything

**Comment by vinceh121** [comment]
I think it would be easier to fetch all this data from OpenStreetMap using [Overpass Turbo](https://overpass-turbo.eu/) with this query:

```
[out:json][timeout:60];
nwr["ISO3166-2"]["type"="boundary"];
out tags;
```

Although I am unsure how complete it is, or if the licensing is compatible with the LGPL.

**Comment by stefangabos** [comment]
this looks really nice and I can further optimise that query. not sure about the licensing though. [this](https://www.openstreetmap.org/copyright) is the OpenStreetMap's license page and it says "If you alter or build upon our data, you may distribute the result only under the same licence". Overall, I think GNU GPL is not compatible with the ODbL license but what we can do though is strictly have the subdivisions data released under the ODbL license, alongside obviously crediting OpenStreetMap for the data source. i am not sure where that stands from a legal point of view

**Comment by stefangabos** [comment]
I rebuild all the subdivisions data and all should be good now
